### ISSUE TITLE
[Idea] Create a new CRD for Kritis Configuration Policy.

### ISSUE THREAD

**Issue by tejal29** [grafeas/kritis#224]
This CRD will define a few configs for Kritis like

1. Which metadata api to use
2. What shd be the default fallback strategy in case Metadata API is not available.
3. Default Cron job run interval.
And anything else which is hardcoded.
This CRD will be cluster scoped for now meaning it will apply to the whole cluster.

The controller and cron job will first fetch this CRD use that instead of defaults.

Pros
1. All main falgs can be customized
2. cluster Admins can apply new config changes without needing to restart kritis.
3. Easy to switch back and forth between  Metadata api
4. Easy to change fallback policy.

**Comment by tejal29** [comment]
Tutorial for adding 
1. CRDs is [here](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)  
2. code generation for Custom Resources is [here](https://blog.openshift.com/kubernetes-deep-dive-code-generation-customresources/)

Now lets add a new custom resource definition
1. Define a new Custom Resource Definition for for ` KritisConfig` like [this](https://github.com/grafeas/kritis/blob/493cd0d4f73bfc6aa8804951c1e6cc8da380c336/artifacts/build-policy-crd.yaml) lets say you name is as `kritis-config.yaml`
You can see what each field mean [here](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)

2. Once you have all these defined, and code generated, you can test it on your cluster
a.  Deploy your Custom Resource using
```
kubectl apply -f kritis-config.yaml
```
b. See if you can list your newly created custom resource
```
kubectl get <resource_plural_name>
```
The resource_plural_name is mentioned in the CRD config. 
For `BuildPolicy` it is `buildpolicies ` and is mentioned [here](https://github.com/grafeas/kritis/blob/493cd0d4f73bfc6aa8804951c1e6cc8da380c336/artifacts/build-policy-crd.yaml#L10)
c. Once you have created a resource definition, you can now create resources of type `KritisConfig` and deploy it using `kubectl apply`
Look here for examples [here](https://github.com/grafeas/kritis/tree/493cd0d4f73bfc6aa8804951c1e6cc8da380c336/artifacts/examples) .
d . if you run `kubectl get` again, you shd be able to see this new resource.


Once you have defined a new Custom Resource, you now will have to generate all code for it so that Kritis web server can access this Config. To do that
1. For Adding a new `CustomResource`, please place the definition [here](https://github.com/grafeas/kritis/tree/master/pkg/kritis/apis/kritis/v1beta1)
Right now, we have all our Custom Resources defined in one file [types.go](https://github.com/grafeas/kritis/blob/master/pkg/kritis/apis/kritis/v1beta1/types.go)
Either create a new file or add to existing one.

You will have to define a Struct sat `KritisConfig` like [this](https://github.com/grafeas/kritis/blob/master/pkg/kritis/apis/kritis/v1beta1/types.go#L65) with tags used by the code generator.
```
// +genclient
// +genclient:noStatus
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
The fields `metav1.TypeMeta` and `metav1.ObjectMeta` are required by the K8 API and specifies the custom resource metadata.
The `Spec` field contains the actual definition in this case it will be a list of Configurations.

The struct name should match the `Kind` field you defined in Step 1. 
e.g. [`BuildPolicy`](https://github.com/grafeas/kritis/blob/493cd0d4f73bfc6aa8804951c1e6cc8da380c336/artifacts/build-policy-crd.yaml#L10) matches struct name [here](https://github.com/grafeas/kritis/blob/master/pkg/kritis/apis/kritis/v1beta1/types.go#L65) 


2. Once you add Custom Resource for `KritisConfig`, you need to define `KritisConfigList` which is just a list of custom resource you just defined. 

3. Finally, you can deploy kritis with this new type and add code something like [this](https://github.com/grafeas/kritis/blob/493cd0d4f73bfc6aa8804951c1e6cc8da380c336/pkg/kritis/crd/securitypolicy/securitypolicy.go#L38) to fetch the config programmatically

**Comment by aysylu** [comment]
#310 will fully address the work started in PR #307. The scope of this issue has been addressed by #301 and #303.
